translator/translate/otel/receiver/jmx/translator_test.go

```diff
@@ -60,8 +65,11 @@ func TestTranslator(t *testing.T) {
 				require.NotNil(t, got)
 				gotCfg, ok := got.(*jmxreceiver.Config)
 				require.True(t, ok)
-				wantCfg := factory.CreateDefaultConfig()
+				wantCfg := factory.CreateDefaultConfig().(*jmxreceiver.Config)
 				require.NoError(t, component.UnmarshalConfig(testCase.want, wantCfg))
+				if wantCfg.ResourceAttributes != nil && wantCfg.ResourceAttributes[hostnameTag] == hostnameTag {
```

Comment:
Hostname must be the same by the translator hostname. Hostname can only be known at runtime and can't be static

Reply:
Can stub out `os.Hostname` in the translator.